src/error.rs

```diff
@@ -0,0 +1,12 @@
+use std::fmt::Display;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct EmptyString;
```

Comment:
`EmptyString` gives the impression that it's a `String` value or something, we should probably append it with "Error", `EmptyStringError`, such that it's clear that it's an error.